discovery/puppetdb/puppetdb.go

```diff
@@ -158,11 +170,11 @@ func NewDiscovery(conf *SDConfig, logger log.Logger, reg prometheus.Registerer)
 
 	d.Discovery = refresh.NewDiscovery(
 		refresh.Options{
-			Logger:   logger,
-			Mech:     "http",
-			Interval: time.Duration(conf.RefreshInterval),
-			RefreshF: d.refresh,
-			Registry: reg,
+			Logger:              logger,
+			Mech:                "http",
```

Comment:
I think `Mech` should be set to `puppetdb` instead of `http`.
`Mech` was set to `http` in #8883, the pull request which originally added the Puppet SD.
@roidelapluie, could you please confirm if `Mech` should be set to "puppetdb"? It'd be better to do it in a separate PR though.

Reply:
Reminder for @roidelapluie that this is still on open question.

Reply:
I raised another PR for this:
https://github.com/prometheus/prometheus/pull/13447